src/ndb/ndb_ephemeris.c

```diff
@@ -0,0 +1,273 @@
+/*
+ * Copyright (C) 2016 Swift Navigation Inc.
+ * Contact: Roman Gezikov <rgezikov@exafore.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+#include <string.h>
+#include <assert.h>
+#include "ndb.h"
+#include "ndb_internal.h"
+#include <libswiftnav/constants.h>
+#include "libswiftnav/logging.h"
+#include "timing.h"
+#include "sbp.h"
+#include "sbp_utils.h"
+
+#define NDB_EPHE_FILE_NAME   "ephe"
+
+static ephemeris_t ndb_ephemeris[PLATFORM_SIGNAL_COUNT] _CCM;
+static ndb_element_metadata_t ndb_ephemeris_md[PLATFORM_SIGNAL_COUNT] _CCM;
+static ndb_file_t ndb_ephe_file = {
+    .name = NDB_EPHE_FILE_NAME,
+    .fh = -1,
+    .expected_size =
+          sizeof(ephemeris_t) * PLATFORM_SIGNAL_COUNT
+        + sizeof(ndb_element_metadata_nv_t) * PLATFORM_SIGNAL_COUNT
+        + sizeof(ndb_file_end_mark),
+    .data_size = sizeof(ephemeris_t),
+    .n_elements = PLATFORM_SIGNAL_COUNT,
+};
+
+typedef struct {
+  ephemeris_t ephe;
+  bool used;
+  ndb_timestamp_t received_at;
+} ephemeris_candidate_t;
+
+typedef enum {
+  EPHE_IDENTICAL,
+  EPHE_NEW_CANDIDATE,
+  EPHE_NEW_TRUSTED,
+  EPHE_CAND_MISMATCH,
+} ndb_ephemeris_status_t;
+
+#define EPHE_CAND_LIST_LEN (MAX_CHANNELS*2/3)
```

Comment:
Just to save memory. Actually, with latest improvement memory isn't a concern, so I change it to MAX_CHANNELS.